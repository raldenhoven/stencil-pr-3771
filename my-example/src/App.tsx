import {MyComponent} from "./components/stencil-generated";

function App() {

  return (
    <div className="App">
        <MyComponent first={'A'} middle={'bit'} last={'broken'}/>
    </div>
  )
}

export default App
