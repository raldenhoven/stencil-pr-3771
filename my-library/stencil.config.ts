import { Config } from '@stencil/core';
import {reactOutputTarget as react} from "@stencil/react-output-target";

export const config: Config = {
  namespace: 'my-library',
  extras: {
    experimentalImportInjection: true,
  },
  outputTargets: [
    react({
      componentCorePackage: 'my-library',
      proxiesFile: '../my-example/src/components/stencil-generated/index.ts',
      includeDefineCustomElements: true
    }),
    {
      type: 'dist',
      esmLoaderPath: '../loader',
    },
    {
      type: 'dist-custom-elements',
    }
  ],
};
