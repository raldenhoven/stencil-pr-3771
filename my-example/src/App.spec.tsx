import {render, screen} from '@testing-library/react'
import App from './App';
import {defineCustomElements} from "my-library/loader";

describe("App", () => {
   let framgent: DocumentFragment;

   beforeEach(() => {
      defineCustomElements(window);
      const result = render(<App/>)
      framgent = result.asFragment();
   })

   it('works', () => {
      expect(framgent).toMatchSnapshot();
   });

   it('injects broken css', () => {
      expect(framgent).toMatchSnapshot();
   });
});
