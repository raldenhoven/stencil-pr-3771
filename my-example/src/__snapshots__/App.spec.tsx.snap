// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App injects broken css 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <my-component
      first="A"
      last="broken"
      middle="bit"
    />
  </div>
</DocumentFragment>
`;

exports[`App works 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <my-component
      first="A"
      last="broken"
      middle="bit"
    />
  </div>
</DocumentFragment>
`;
